### PR TITLE
Get full dimensional polytope for the Python interface

### PR DESCRIPTION
**Pull request by hariszaf**
If applied, this commit will include a new class in the Python interface of _volesti_

The class called ```low_dim_HPolytope``` will allow the user:
* to build a low dimensional polytope
* to get its corresponding full dimensional polytope

For the latter, the ```full_dimensiolal_polytope``` function is used and its output includes in a tuple format the following:
* an object of the ```HPolytope``` class for the case of the full dimensional polytope; 
* the ```shift``` vector
* the ```N``` matrix

### CODE REVIEW

File: volestipy/volestipy/include/bindings.h [hunk 1]
```diff
@@ -27,51 +27,71 @@
...
 
+// The preHPolytopeCPP class is responsible for the preprocess step of the polytope as well as for getting the full dimensional polytope 
+class lowDimHPolytopeCPP{
 
+   public:
+      
+      MT A,Aeq;
+      VT b,beq;
+      
+      std::pair<Point,NT> CheBall;
```
Comment (TolisChal) [inline volestipy/volestipy/include/bindings.h]: Do you need CheBall in low dim polytope?

File: volestipy/volestipy/volestipy.pyx [hunk 2]
```diff
@@ -115,3 +126,86 @@ cdef class HPolytope:
...
+         
+      # likewise, for the A matrix of the full dimensional polytope
+      A_full = np.zeros((k, n_of_cols_in_N), dtype=np.float64, order="C")
+      for i in range(k):
+         for j in range(n_of_cols_in_N):
+            A_full[i,j] = np.asarray(A_full_extra_trans[j,i])
+      del A_full_extra_trans
+      
+      # finally, we need to build an HP object for the full dumensional polytope we got
+      full_dimensional_polytope = HPolytope(A_full,b_full)
```
Comment (TolisChal) [inline volestipy/volestipy/volestipy.pyx:207]: Maybe del b_full after the declaration of object `full_dimensional_polytope`?

File: volestipy/volestipy/volestipy.pyx [hunk 3]
```diff
@@ -115,3 +126,86 @@ cdef class HPolytope:
...
+   cdef double[::1] _beq
+
+# set the specs of the class
+   def __cinit__(self, double[:,::1] A, double[::1] b, double[:,::1] Aeq, double[::1] beq):
+      self._A = A
+      self._b = b
+      self._Aeq = Aeq
+      self._beq = beq
+      n_rows_of_A, n_cols_of_A = A.shape[0], A.shape[1] 
+      n_row_of_Aeq, n_cols_of_Aeq = Aeq.shape[0], Aeq.shape[1]
```
Comment (TolisChal) [inline volestipy/volestipy/volestipy.pyx:149]: Please check if:
1) n_rows_of_A == b.shape[0]
2) n_row_of_Aeq == beq.shape[0]
3) n_cols_of_A == n_cols_of_Aeq

If not raise an exception

File: volestipy/volestipy/include/bindings.h (hunk 1 above)
Comment (hariszaf) [reply volestipy/volestipy/include/bindings.h]: Nope.
I will remove it.

File: volestipy/volestipy/volestipy.pyx (hunk 3 above)
Comment (hariszaf) [reply volestipy/volestipy/volestipy.pyx:149]: Three if statements were added to check for these equalities

File: volestipy/volestipy/volestipy.pyx (hunk 2 above)
Comment (hariszaf) [reply volestipy/volestipy/volestipy.pyx:207]: If we remove ```b_full```, we should remove the ```A_full``` as well right?

**Review by TolisChal — APPROVED** [review]

**Pull request merged**